backend/src/email/services/email-template.service.ts

```diff
@@ -161,10 +165,20 @@ const storeTemplate = async ({
   replyTo,
   from,
 }: StoreTemplateInput): Promise<StoreTemplateOutput> => {
+  if (!client.stripTags(subject).trim() || !client.stripTags(body).trim()) {
+    throw new TemplateError('Message subject or template cannot be empty!')
+  }
   // extract params from template, save to db (this will be done with hook)
+  const sanitizedSubject = client.replaceNewLinesAndSanitize(subject)
+  const sanitizedBody = client.replaceNewLinesAndSanitize(body)
+  if (!sanitizedSubject || !sanitizedBody) {
```

Comment:
Hmm...what is the edge case that this second check is checking for? Given that the first step has already stripped all tags and checked if it is empty, wouldn't the second check be a subset of the first step?

Reply:
I believe you are right, this was the first check I implemented, and then I refined it to add a check to reject templates which may have valid tags but are essentially empty. The second implementation would encapsulate the logic of the former one. Will remove!